tests/yam/agama/validate_headless.pm

```diff
@@ -0,0 +1,32 @@
+## Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Agama headless validataion
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base Yam::Agama::agama_base;
+use strict;
+use warnings;
+use testapi qw(
+  assert_script_run
+  select_console
+  upload_logs
+);
+use utils qw(systemctl);
+
+sub run {
+    select_console 'root-console';
+
+    # verify we are in muti-user and not graphic target.
+    assert_script_run('systemctl list-units --type=target | grep multi-user.target');
+    systemctl('is-active graphical.target', expect_false => 1);
+    systemctl('is-active x11-autologin.service', expect_false => 1);
+    systemctl('is-enabled multi-user.target');
+    systemctl('is-active multi-user.target');
+
+    # check there is not instance of the firefox browser open
+    assert_script_run('! pgrep firefox');
+    assert_script_run('! pgrep gnome-session');
```

Comment:
I don't see any process using the iso for prod in a headful system.

Reply:
removed.

Reply:
In headless mode (multi-user.target), neither Firefox nor IceWM will be launched,so swap gnome-session with icewm-session. cause I found it's showed in a [headful system](https://openqa.suse.de/tests/17511537#step/patch_agama_tests/12).